src/SettingsView/EventType.js

```diff
@@ -1,103 +1,150 @@
-			<Card
-				expanded={this.state.expanded}
-				onExpandChange={this.handleExpandChange}>
-				<CardHeader
-					title={this.props.eventType.Name}
-					subtitle={`${config.descriptions.textNumberOfEvents} ${this.props.eventType.NumberOfEvents}`}
-					actAsExpander={true}
-					showExpandableButton={true}
-				/>
-				<CardText
-					expandable={true}
-					className={css(AppStyles.dFlex)}>
-					<List className={css(AppStyles.w50)}>
-						{queries.map((query) => {
-							return(
-							<EventQueryListItem
-								query={query}
-								deleteQuery={this.props.deleteQuery}
-								key={query.Id}/>);
-							}
-						)}
-					</List>
-					<List className={css(AppStyles.w50)}>
-						{attributes.map((attribute) => { return(
-							<ListItem 
-								primaryText={attribute.Name}
-								key={attribute.Id}
-							/>);
-						})}
-					</List>
-				</CardText>
-			</Card>
-			</div>
-		);
-	}
-}
-
-export default ConnectionComponent.argosConnector()(props => ({
-	queries: config.backendRESTRoute + `/eventtype/${props.eventType.Id}/queries`,
-	attributes: config.backendRESTRoute + `/eventtype/${props.eventType.Id}/attributes`,
-	deleteQuery: query => ({
-		deleteQueryResponse: {
-			url: config.backendRESTRoute + `/eventquery/${query.Id}/delete`,
-			method: 'DELETE',
-			body: "",
-			andThen: () => ({
-				queries: {
-					url: config.backendRESTRoute + `/eventtype/${props.eventType.Id}/queries`,
-					refreshing: true
-				}
-			})
-		}
-	})
+import React from 'react';
+import {Card, CardHeader, CardText } from 'material-ui/Card';
+import {List, ListItem} from 'material-ui/List';
+import EventQueryListItem from  './EventQueryListItem.js';
+import EntityMappingListItem from  './EntityMappingListItem.js';
+import config from './../config/config.js';
+import { Col, Container } from 'react-grid-system';
+import { PromiseState } from 'react-refetch';
+import ConnectionComponent from './../Utils/ConnectionComponent.js';
+import ErrorMessage from './../Utils/ErrorMessage.js';
+import IconButton from 'material-ui/IconButton';
+import IconAdd from 'material-ui/svg-icons/content/add';
+import IconDelete from 'material-ui/svg-icons/action/delete';
+import IconEdit from 'material-ui/svg-icons/editor/mode-edit';
+
+class EventType extends ConnectionComponent {
+	constructor(props) {
+		super(props);
+		this.state = {
+			expanded: false,
+			mappingExpanded: false,
+		};
+		this.handleExpandChange = this.handleExpandChange.bind(this);
+		this.handleMappingsExpandChange = this.handleMappingsExpandChange.bind(this);
+		this.handleEventTypeDeletion = this.handleEventTypeDeletion.bind(this);
+	}
+
+	handleExpandChange(expanded) {
+		this.setState({expanded: expanded});
+	}
+
+    handleMappingsExpandChange(expanded) {
+		this.setState({mappingExpanded: expanded});
+	}
+	
+	handleEventTypeDeletion() {
+		this.props.deleteEventType(this.props.eventType);
+	}
+
+	getEventTypeHeaderButtons() {
+		return [
+			// TODO add functionality

```

Comment:
![INFO](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-info.png 'Severity: INFO') Complete the task associated to this TODO comment. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](http://bpt-lab.org/sonarqube/coding_rules#rule_key=javascript%3AS1135)